binr/rax2/rax2.c

```diff
@@ -228,8 +246,44 @@ static int rax(char *str, int len, int last) {
 		return true;
 	}
 	if (flags & (1 << 2)) { // -S
+		int i, dec_val;
+		printf("%s\n", str);
 		for (i = 0; i < len; i++) {
-			printf ("%02x", (ut8) str[i]);
+			if (str[i] == '\\') {
+         		++i; 
+				if ((dec_val = escape_to_dec (str[i])) != 2) {
+					printf ("%02x", dec_val);
+				} else if (str[i] == 'x') {	// '\xhhhh..'
+					i++;
+					while (IS_HEXCHAR (str[i])) {
+						printf ("%c", tolower (str[i]));
+						i++;
+					}
+					i--; 
+				} else if (IS_OCTAL (str[i])) {
+					if (!IS_OCTAL (str[i + 1])) {	// '\o'
+						printf ("0%c", str[i]);
+					} else if ((IS_OCTAL (str[i + 1])) && !IS_OCTAL (str[i + 2])) { 	// '\oo'
+						char octal_str[3], *modified_str;
+						strncpy (octal_str, str + i, 2);
+						i++;
+						modified_str = r_str_newf (((*octal_str != '0') ? "0%s" : ""), octal_str);
+						ut64 n = r_num_math (num, modified_str); 
+						printf ("%02x", (ut8) n);
+					} else {	// '\ooo'
+						char octal_str[4], *modified_str;
+						strncpy (octal_str, str + i, 3);
+						i += 2;;
+						modified_str = r_str_newf (((*octal_str != '0') ? "0%s" : ""), octal_str);
+						ut64 n = r_num_math (num, modified_str);
+						((ut16) n) > 255 ? (printf ("%03x", (ut16) n)) : (printf ("%02x", (ut16) n));
```

Comment:
dont use ternary operator for calls, here you just want it to chagne the format string